featuretools/tests/entityset_tests/test_entity.py

```diff
@@ -158,3 +160,62 @@ def test_variable_types_unmodified():
                              time_index='transaction_time',
                              variable_types=variable_types)
     assert old_variable_types == variable_types
+
+
+def test_passing_strings_to_variable_types_entity_init():
+    variable_types = find_variable_types()
+    reversed_variable_types = {str(v): k for k, v in variable_types.items()}
+    reversed_variable_types['unknown variable'] = 'some unknown type string'
+
+    es = EntitySet()
+    dataframe = pd.DataFrame(columns=list(reversed_variable_types))
+    entity = Entity('reversed_variable_types', dataframe, es,
+                    variable_types=reversed_variable_types,
+                    index="<class 'featuretools.variable_types.variable.Index'>",
+                    time_index="<class 'featuretools.variable_types.variable.DatetimeTimeIndex'>",
+                    )
+    for variable in entity.variables:
+        description = variable.to_data_description()
+        _variable = deserialize.description_to_variable(description, entity=entity)
+        assert variable.__eq__(_variable)
```

Comment:
this test should use the same approach as `test_passing_strings_to_variable_types_from_dataframe`